modules/kubernetes-addons/cluster-autoscaler/main.tf

```diff
@@ -1,15 +1,94 @@
+locals {
+  name            = try(var.helm_config.name, "cluster-autoscaler")
+  namespace       = try(var.helm_config.namespace, "kube-system")
+  service_account = "${local.name}-sa"
+}
+
 module "helm_addon" {
-  source            = "../helm-addon"
+  source = "../helm-addon"
+
   manage_via_gitops = var.manage_via_gitops
-  set_values        = local.set_values
-  helm_config       = local.helm_config
-  irsa_config       = local.irsa_config
-  addon_context     = var.addon_context
+
+  helm_config = merge({
+    name        = local.name
+    chart       = local.name
+    version     = "9.19.1"
+    repository  = "https://kubernetes.github.io/autoscaler"
+    namespace   = local.namespace
+    description = "Cluster AutoScaler helm Chart deployment configuration."
+    values = [templatefile("${path.module}/values.yaml", {
+      aws_region     = var.addon_context.aws_region_name,
+      eks_cluster_id = var.addon_context.eks_cluster_id
+    })]
+    },
+    var.helm_config
+  )
+
+  set_values = [
+    {
+      name  = "rbac.serviceAccount.create"
+      value = "false"
+    },
+    {
+      name  = "rbac.serviceAccount.name"
+      value = local.service_account
+    },
+    {
+      name  = "image.tag"
+      value = "v${var.eks_cluster_version}.0"
```

Comment:
this is the magic sauce - needs to be in the form of `v#.#.#` and using a default of `v#.#.0` since it will always exist (not sure how we could pull the latest patch version for the cluster version )